aptos-move/framework/aptos-framework/sources/account.move

```diff
@@ -495,6 +512,36 @@ module aptos_framework::account {
         option::swap_or_fill(&mut account_resource.rotation_capability_offer.for, recipient_address);
     }
 
+    /// For the given account, add an entry to `OriginatingAddress` table mapping the account's
+    /// authentication key to the account's address.
+    ///
+    /// Can be used as a followup to `rotate_authentication_key_call()` to reconcile the
+    /// `OriginatingAddress` table, or to establish a mapping for a new account that has not yet had
+    /// its authentication key rotated.
+    ///
+    /// Aborts if there is already an entry in the `OriginatingAddress` table for the account's
+    /// authentication key.
+    ///
+    /// Kept as a private entry function to ensure that after an unproven rotation via
+    /// `rotate_authentication_key_call()`, the `OriginatingAddress` table is only updated under the
+    /// authority of the new authentication key.
+    entry fun set_originating_address(account: &signer) acquires Account, OriginatingAddress {
+        let account_addr = signer::address_of(account);
+        assert!(exists<Account>(account_addr), error::not_found(EACCOUNT_DOES_NOT_EXIST));
+        let auth_key_as_address =
+            from_bcs::to_address(borrow_global<Account>(account_addr).authentication_key);
+        let address_map_ref_mut =
+            &mut borrow_global_mut<OriginatingAddress>(@aptos_framework).address_map;
+        if (table::contains(address_map_ref_mut, auth_key_as_address)) {
+            assert!(
```

Comment:
Is this assert necessary if there are no changes?

Reply:
I think the important piece of info that you may have missed based on another one of your review comments is that the `OriginatingAddress` table is only one-to-one: this is so that if you have a public key you can reverse lookup which account is verifiably linked with that auth key, and only approve a txn for that account, deterministically.

In this case, this assert is to enforce that the one-to-one table mapping is not disturbed by setting a new originating address following an unverified rotation